code/game/data_huds.dm

```diff
@@ -436,52 +436,47 @@
 /*~~~~~~~~~~~~
 	Circutry!
 ~~~~~~~~~~~~~*/
-/obj/item/electronic_assembly/proc/diag_hud_set_circuithealth(hide = FALSE)
+/atom/proc/diag_hud_set_circuithealth(percent, hide = FALSE)
```

Comment:
Is it an issue that this proc is on atom? I had to put it there since the component can be attached to any atom

Reply:
You should be doing this code from inside the component